.github/workflows/ci.yml

```diff
@@ -2,56 +2,63 @@ name: CI
 
 on:
   pull_request:
-    branches:
-      - master
   push:
-    branches:
-      - master
+    branches: [master]
 
 jobs:
-  setup:
+  test:
+    runs-on: ubuntu-20.04
+
     env:
       MIX_ENV: test
 
     strategy:
-      fail-fast: false
+      # https://hexdocs.pm/elixir/compatibility-and-deprecations.html#between-elixir-and-erlang-otp
       matrix:
-        os: [ubuntu-22.04, ubuntu-20.04]
-        elixir_version: [1.13, 1.14, 1.15, 1.16]
-        otp_version: [24, 25, 26]
+        elixir: [1.16, 1.15, 1.14, 1.13]
+        otp: [26, 25, 24, 23]
+        include:
+          - elixir: 1.17
+            otp: 27
         exclude:
-          - otp_version: 26
-            elixir_version: 1.13
-
-    runs-on: ${{ matrix.os }}
+          - elixir: 1.15
+            otp: 23
+          - elixir: 1.16
+            otp: 23
+          - elixir: 1.13
+            otp: 26
 
     steps:
       - uses: actions/checkout@v4
 
       - uses: erlef/setup-beam@v1
         with:
-          otp-version: ${{matrix.otp_version}}
-          elixir-version: ${{matrix.elixir_version}}
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
 
       - uses: actions/cache@v4
         with:
           path: |
             deps
             _build
-          key: deps-${{ runner.os }}-${{ matrix.otp_version }}-${{ matrix.elixir_version }}-${{ hashFiles('**/mix.lock') }}
+          key: test-otp-${{ matrix.otp }}-elixir-${{ matrix.elixir }}-ref-${{ github.head_ref || github.ref }}-mix-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            deps-${{ runner.os }}-${{ matrix.otp_version }}-${{ matrix.elixir_version }}
+            test-otp-${{ matrix.otp }}-elixir-${{ matrix.elixir }}-ref-${{ github.head_ref || github.ref }}-mix-
+            test-otp-${{ matrix.otp }}-elixir-${{ matrix.elixir }}-ref-refs/heads/master-mix-
 
       - run: mix deps.get
-
-      - run: mix format --check-formatted
-
       - run: mix deps.unlock --check-unused
-
       - run: mix deps.compile
-
       - run: mix compile --warnings-as-errors
-
       - run: mix credo --strict --format=oneline
-
       - run: mix test --warnings-as-errors --cover
+
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: erlef/setup-beam@v1
+        with:
+          elixir-version: 1
+          otp-version: 27
+      - run: mix format --check-formatted
```

Comment:
Does the `with` block here work?

As I understand it, the `run:` key is used to specify a shell command or script to be executed, and it does not accept the `with:` field. The `with:` field is typically used in the context of actions that require input parameters, but not directly in a `run:` step.

I've tried to set up something similar with dialyzer and my editor warns me that I can't. ChatGPT also tells me this isn't allowed. So I'm curious if this works for you here.

Reply:
👋 @cliftonmcintosh 

I might be misunderstanding your question, but why would `with` be used here?

Reply:
If your question is about the `with` on `uses: erlef/setup-beam@v1` then yes, it works.

Reply:
NVM. I see that it is valid because the with block is for the setup step